src/main/java/com/limechain/runtime/hostapi/HostApi.java

```diff
@@ -0,0 +1,193 @@
+        HostApi.runtime = runtime;
+    }
+
+    private static Level getLogLevel(int i) {
+        return switch (i) {
+            case 0 -> Level.SEVERE;
+            case 1 -> Level.WARNING;
+            case 2 -> Level.INFO;
+            case 3 -> Level.FINE;
+            default -> Level.FINEST;
+        };
+    }
+
+    public static int extAllocatorMallocVersion1(int size) {
+        Memory memory = getMemory();
+        ByteBuffer buffer = getByteBuffer(memory);
+        int position = buffer.position();
+        if (size > buffer.limit() - position) {
+            memory.grow(buffer.limit() - position);
+        }
+        buffer.position(position + size);
+
+        return position;
+    }
+
+    private static ByteBuffer getByteBuffer(Memory memory) {
+        ByteBuffer buffer;
+        try {
+            Field bufferField = memory.getClass().getDeclaredField("buffer");
+            bufferField.setAccessible(true);
+            buffer = (ByteBuffer) bufferField.get(memory);
+        } catch (IllegalAccessException | NoSuchFieldException e) {
+            log.log(Level.WARNING, "Unable to get buffer", e);
+            buffer = null;
+        }
+        return buffer != null ? buffer : memory.buffer();
+    }
+
+    public static int extHashingTwox64Version1(long addr) {
+        byte[] dataToHash = getDataFromMemory(addr);
+
+        byte[] hash0 = hash64(0, dataToHash);
+
+        return putDataToMemory(hash0);
+    }
+
+    public static int extHashingTwox128Version1(long addr) {
+        byte[] dataToHash = getDataFromMemory(addr);
+
+        byte[] hash0 = hash64(0, dataToHash);
+        byte[] hash1 = hash64(1, dataToHash);
+
+        ByteBuffer buffer = ByteBuffer.allocate(16);
+        buffer.put(hash0);
+        buffer.put(hash1);
+
+        byte[] byteArray = buffer.array();
+        return putDataToMemory(byteArray);
+    }
+
+    public static int extHashingTwox256Version1(long addr) {
+        byte[] dataToHash = getDataFromMemory(addr);
+
+        byte[] hash0 = hash64(0, dataToHash);
+        byte[] hash1 = hash64(1, dataToHash);
+        byte[] hash2 = hash64(2, dataToHash);
+        byte[] hash3 = hash64(3, dataToHash);
+
+        ByteBuffer buffer = ByteBuffer.allocate(32);
+        buffer.put(hash0);
+        buffer.put(hash1);
+        buffer.put(hash2);
+        buffer.put(hash3);
+
+        byte[] byteArray = buffer.array();
+        return putDataToMemory(byteArray);
+    }
+
+    private static byte[] hash64(int seed, byte[] dataToHash) {
+        final long res3 = LongHashFunction
+                .xx(seed)
+                .hashBytes(dataToHash.clone());
+
+        final ByteBuffer buffer = ByteBuffer
+                .allocate(8)
+                .order(ByteOrder.LITTLE_ENDIAN)
+                .putLong(res3);
+
+        return buffer.array();
+    }
+
+    public static void extAllocatorFreeVersion1(int i) {
+        //TODO: Try marking the part of the bytebuffer as free?
+    }
```

Comment:
Function with empty body. Either throw unimplemented error or remove it. Also please be more specific about the comment. I don't quite understand it.